moto/logs/models.py

```diff
@@ -366,13 +366,35 @@ def filter_log_events(
         if interleaved:
             events = sorted(events, key=lambda event: event["timestamp"])
 
-        if next_token is None:
-            next_token = 0
+        first_index = 0
+        if next_token:
+            try:
+                group, stream, event_id = next_token.split("/")
+                if group != log_group_name:
+                    raise ValueError()
```

Comment:
Hi @neil-greenwood, thanks for the PR! Can we have a test for this error scenario as well?

Reply:
Sure.